test/e2e/common_test.go

```diff
@@ -291,15 +294,41 @@ func PodmanTestCreateUtil(tempDir string, remote bool) *PodmanTestIntegration {
 	}
 	if remote {
 		p.PodmanTest.RemotePodmanBinary = podmanRemoteBinary
-		uuid := stringid.GenerateNonCryptoID()
+		var pathPrefix string
 		if !rootless.IsRootless() {
-			p.RemoteSocket = fmt.Sprintf("unix:/run/podman/podman-%s.sock", uuid)
+			pathPrefix = "/run/podman/podman"
 		} else {
 			runtimeDir := os.Getenv("XDG_RUNTIME_DIR")
-			socket := fmt.Sprintf("podman-%s.sock", uuid)
-			fqpath := filepath.Join(runtimeDir, socket)
-			p.RemoteSocket = fmt.Sprintf("unix:%s", fqpath)
+			pathPrefix = filepath.Join(runtimeDir, "podman")
 		}
+		// We want to avoid collisions in socket paths, but using the
+		// socket directly for a collision check doesn’t work; bind(2) on AF_UNIX
+		// creates the file, and we need to pass a unique path now before the bind(2)
+		// happens. So, use a podman-%s.sock-lock empty file as a marker.
+		tries := 0
+		for {
+			uuid := stringid.GenerateNonCryptoID()
+			lockPath := fmt.Sprintf("%s-%s.sock-lock", pathPrefix, uuid)
+			lockFile, err := os.OpenFile(lockPath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0700)
+			if err == nil {
+				lockFile.Close()
+				p.RemoteSocket = fmt.Sprintf("unix:%s-%s.sock", pathPrefix, uuid)
```

Comment:
Don't you need you need an `unlink` at some (deferred) point?

Reply:
I can’t see any pre-existing code that removes the `.sock` files, so this adds no code that removes the `.sock-lock` files either.

Per the PR description, I am assuming that this runs in a VM that is going to be torn down; is that incorrect?

Reply:
I've been known to run the `e2e` tests on my laptop. They're an enormous PITA to run, very slow and cumbersome, but it's necessary sometimes. (I can't say that I've looked for stray files in `/run`, though. And I'm not willing to spend an hour right now to find out). Don't the socket files get cleaned up once the controlling process exits?

Reply:
Update: yes. `podman system service --timeout=0 /tmp/podman-foo.sock` creates podman-foo.sock, then when I ^C the .sock file is gone.

Reply:
So I checked, and it does seem we "should" remove the remote socket, assuming this function is actually (and reliably) called:  

https://github.com/containers/podman/blob/4105b025d60dd6d067fa293de11c45dcfd7ea30e/test/e2e/libpod_suite_remote_test.go#L137-L140

Note: This might also be a good place for some more logging, in case where removal is requested but the socket file is unexpectedly not there.

Reply:
never-mind, I'm dumb, there is logging.

Reply:
> Update: yes. `podman system service --timeout=0 /tmp/podman-foo.sock` creates podman-foo.sock, then when I ^C the .sock file is gone.

I’m afraid can’t see how that happens — but `podman system service` wouldn’t be in charge of the test-only lock file; I have updated the place @cevich found to also remove the lock.